### PR TITLE
TEST: groupby(as_index=False, sort=False).aggregate formerly (?) gave unexpected results with a list-like function

### LINKED ISSUE #18473: test needed: groupby(as_index=False, sort=False).aggregate formerly (?) gave unexpected results with a list-like function
#### Code Sample, a copy-pastable example if possible

```python
df = pd.DataFrame({
  'group' : [str(x) for x in range(12)],
  'file'  : [str(x) for x in range(12)]})
#Out[20]: 
#   file group
#0     0     0
#1     1     1
#2     2     2
#3     3     3
#4     4     4
#5     5     5
#6     6     6
#7     7     7
#8     8     8
#9     9     9
#10   10    10
#11   11    11
df.groupby('group', as_index=False, sort=False).aggregate({ 'file' : lambda file : list(file)})
#Out[21]: 
#   group  file
#0      0   [0]
#1      1   [1]
#2      2  [10]
#3      3  [11]
#4      4   [2]
#5      5   [3]
#6      6   [4]
#7      7   [5]
#8      8   [6]
#9      9   [7]
#10    10   [8]
#11    11   [9]
```
#### Problem description

The aggregate obviously hasn't aggregated the correct thing, but rather seems to have assumed some incorrect sorting of the rows.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-87-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_CA.UTF-8
LOCALE: en_CA.UTF-8

pandas: 0.21.0
pytest: 3.2.2
pip: 9.0.1
setuptools: 28.8.0
Cython: 0.26.1
numpy: 1.13.3
...
</details>

### PR DESCRIPTION
**Pull request by discort**
- [x] closes #18473
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/groupby/aggregate/test_other.py [hunk 1]
```diff
@@ -512,3 +512,20 @@ def test_agg_category_nansum(observed):
...
+    df = pd.DataFrame({'A': [str(x) for x in range(10)],
+                       'B': [str(x) for x in range(10)]})
+    result = df.groupby('A', as_index=False, sort=False)\
+               .aggregate({'B': lambda x: list(x)})
+    expected = df.copy()
+    expected['B'] = df['B'].apply(lambda x: list(x))
+    tm.assert_frame_equal(result, expected)
+
+    # Testing normal func
+    result = df.groupby('A', as_index=False, sort=False)\
```
Comment (WillAyd) [inline pandas/tests/groupby/aggregate/test_other.py]: Is this not already tested elsewhere? No need to duplicate if so

File: pandas/tests/groupby/aggregate/test_other.py [hunk 2]
```diff
@@ -512,3 +512,20 @@ def test_agg_category_nansum(observed):
...
+
+
+def test_agg_list_like_func():
+    # GH 18473
+    df = pd.DataFrame({'A': [str(x) for x in range(10)],
+                       'B': [str(x) for x in range(10)]})
+    result = df.groupby('A', as_index=False, sort=False)\
+               .aggregate({'B': lambda x: list(x)})
+    expected = df.copy()
+    expected['B'] = df['B'].apply(lambda x: list(x))
```
Comment (WillAyd) [inline pandas/tests/groupby/aggregate/test_other.py]: Would be preferable to just manually construct the expected output here for both readability and risk mitigation. A smaller example (say `range(3)`) should suffice if it makes things easier

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by discort** [comment]
@WillAyd

File: pandas/tests/groupby/aggregate/test_other.py [hunk 3]
```diff
@@ -512,3 +512,14 @@ def test_agg_category_nansum(observed):
     if observed:
         expected = expected[expected != 0]
     tm.assert_series_equal(result, expected)
+
+
+def test_agg_list_like_func():
+    # GH 18473
+    df = pd.DataFrame({'A': [str(x) for x in range(3)],
+                       'B': [str(x) for x in range(3)]})
+    result = df.groupby('A', as_index=False, sort=False)\
```
Comment (WillAyd) [inline pandas/tests/groupby/aggregate/test_other.py]: Can you use implicit line breakage here instead?

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
@jreback not sure nothing logically sticks out. Would have to bisect to see

**Comment by discort** [comment]
@WillAyd

**Comment by WillAyd** [comment]
@discort can you merge in from master to see if that fixes the CircleCI failure?

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

**Comment by jreback** [comment]
thanks!

**Pull request merged**